tagbotgitlab/tagbot.py

```diff
@@ -65,13 +66,25 @@ def handle_open(payload):
     p = client.projects.get(p_id, lazy=True)
     mr_id = get_in(payload, "changes", "iid", "current")
     mr = p.mergerequests.get(mr_id, lazy=True)
+
     print("Approving MR")
     mr.approve()
+
     # Add printing the MR state to assist in debugging cases where the mr.merge() below
     # returns an error
     mr = p.mergerequests.get(mr_id, lazy=False)
     print(mr)
-    print("Merging MR")
+
+    # Add timeout to wait for the head pipeline to be associated properly with the MR
+    # Avoids merge failures
+    timeout = 1
+    while timeout <= 3 and mr.head_pipeline is None:
+        print(f"Sleeping for {timeout} seconds")
+        time.sleep(timeout)
```

Comment:
This is going to sleep for 1 second, then 2, then 3 was this intended? I guess its kind of like an exponential timeout type of thing?

Reply:
Yeah, it was intended to work as a bootleg exponential backoff